gateway-modules/apim-ballerina-native-component/src/test/java/org/wso2/carbon/apimgt/ballerina/util/ListFilesTestCase.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.ballerina.util;
+
+import org.ballerinalang.model.values.BStringArray;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.codegen.ProgramFile;
+import org.ballerinalang.util.program.BLangFunctions;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.apimgt.ballerina.caching.util.BTestUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Cache Test class which handles caching test for ballerina native cache implementation for API Manager
+ *
+ * @since 0.10-SNAPSHOT
+ */
+
+public class ListFilesTestCase {
+
+    private ProgramFile bLangProgram;
+    final String home = System.getProperty("user.home");
+
+    @BeforeClass
+    public void setup() {
+
+
+//        //      /home/sabeena
+//        File notADir = new File(home + "/notADir");
+//        notADir.delete();
+//
+//        //boolean allJsonDirCreated = (new File("/home/sabeena/Desktop/allJsonDir")).mkdirs();
+//        boolean allJsonDirCreated = (new File(home + "/allJsonDir")).mkdirs();
+//        boolean noJsonDirCreated = (new File(home + "/noJsonDir")).mkdirs();
+//
+//        if (allJsonDirCreated) {
+//            File a = new File(home + "/allJsonDir/a.json");
+//            File b = new File(home + "/allJsonDir/b.json");
+//
+//            try {
+//                a.createNewFile();
+//                b.createNewFile();
+//            } catch (IOException ioe) {
+//            }
+//        }
+//
+//        if (noJsonDirCreated) {
+//            File a = new File(home + "/noJsonDir/a.txt");
+//            File b = new File(home + "/noJsonDir/b.txt");
+//
+//            try {
+//                a.createNewFile();
+//                b.createNewFile();
+//            } catch (IOException ioe) {
+//            }
+//        }
+
+        //      /home/sabeena
```

Comment:
Shall we clean up if these commented code blocks are not needed?

Reply:
okay